framework/src/util/AnyMap.cpp

```diff
@@ -892,6 +863,56 @@ any_map::unordered_any_cimap& any_map::uoci_m()
   return *map.uoci;
 }
 
+void any_map::copy_from(const any_map& other)
+{
+  switch (other.type) {
+    case map_type::ORDERED_MAP:
+      map.o = new ordered_any_map(other.o_m());
+      break;
+    case map_type::UNORDERED_MAP:
+      map.uo = new unordered_any_map(other.uo_m());
+      break;
+    case map_type::UNORDERED_MAP_CASEINSENSITIVE_KEYS:
+      map.uoci = new unordered_any_cimap(other.uoci_m());
+      break;
+    default:
+      throw std::logic_error("invalid map type");
+  }
+}
+
+void any_map::move_from(any_map&& other) noexcept
+{
+  switch (other.type) {
+    case map_type::ORDERED_MAP:
+      map.o = other.map.o;
+      other.map.o = nullptr;
```

Comment:
This seems safer but why do we care to perform this operation if the object is supposed to be discarded anyway? If the user tries to use the object after it has been moved, it's a logic error in the program. There is no guarantee that the object checks for null before performing an operation on it's member pointer anyway.

Don't pay for what you don't need to.

Reply:
`o` is a pointer to the map. If this assignment to `nullptr` isn't done, the temporary `AnyMap`'s destructor will delete the map pointed to by `o`. Hence assigning this to `nullptr` so that the deletion doesn't occur.

Reply:
Based on offline discussion, we both felt that the underlying pointers should be `unique_ptr`s. That will make the code more compact and idiomatic.